contracts/Registry.sol

```diff
@@ -0,0 +1,182 @@
+import "@openzeppelinV3/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelinV3/contracts/utils/Address.sol";
+import "@openzeppelinV3/contracts/utils/EnumerableSet.sol";
+
+import {VaultAPI} from "./BaseStrategy.sol";
+
+enum VaultStatus {Invalid, Experimental, Endorsed, Deprecated}
+
+struct VaultData {
+    VaultStatus status;
+    address vaultAddress;
+    string vaultName;
+    string vaultSymbol;
+    string vaultApiVersion;
+    address tokenAddress;
+    string tokenName;
+    string tokenSymbol;
+    uint256 tokenDecimals; // NOTE: Vault.decimals() == Token.decimals()
+}
+
+contract Registry {
+    using Address for address;
+    using SafeMath for uint256;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    address public governance;
+    address pendingGovernance;
+
+    EnumerableSet.AddressSet private vaults;
+    mapping(address => address) public latestVaultForToken;
+
+    // Keep track of status
+    EnumerableSet.AddressSet private experimentalVaults;
+    EnumerableSet.AddressSet private endorsedVaults;
+    EnumerableSet.AddressSet private deprecatedVaults;
+
+    event VaultUpgrade(address indexed token, address indexed oldVault, address newVault);
+
+    constructor(address _governance) public {
+        require(_governance != address(0), "Missing Governance");
+        governance = _governance;
+    }
+
+    // Governance setters
+    function setGovernance(address _governance) external onlyGovernance {
+        pendingGovernance = _governance;
+    }
+
+    function acceptGovernance() external {
+        require(msg.sender == pendingGovernance, "!pendingGovernance");
+        governance = msg.sender;
+    }
+
+    modifier onlyGovernance {
+        require(msg.sender == governance, "!governance");
+        _;
+    }
+
+    // Vault set actions
+    function addVault(address _vault, bool experimental) public onlyGovernance {
+        require(_vault.isContract(), "Vault is not a contract");
+        // Checks if vault is already on the array
+        require(!vaults.contains(_vault), "Vault already exists");
+        // Adds unique _vault to vaults array
+        vaults.add(_vault);
+
+        if (experimental) {
+            experimentalVaults.add(_vault);
+        } else {
+            endorsedVaults.add(_vault);
+        }
+
+        // Keep track of latest Vault (newest is considered best)
+        // NOTE: A vault should only be added to the registry when it's considered
+        //       "production" ready, signaling it as the one people should upgrade to
+        address token = VaultAPI(_vault).token();
+        address oldVault = latestVaultForToken[token];
+        latestVaultForToken[token] = _vault;
+        emit VaultUpgrade(token, oldVault, _vault);
+    }
+
+    function endorseVault(address _vault) public onlyGovernance {
+        require(experimentalVaults.contains(_vault), "!experimental");
+        experimentalVaults.remove(_vault);
+        endorsedVaults.add(_vault);
```

Comment:
I think you should emit an event after adding the vault in the endorsedVaults list. Especially when you move the vault from "experimental" to "endorsed".